CMakeLists.txt

```diff
@@ -0,0 +1,179 @@
+         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  PRIVATE $<BUILD_INTERFACE:
+          ${CMAKE_CURRENT_SOURCE_DIR}/include
+          ${CMAKE_CURRENT_SOURCE_DIR}/celt
+          ${CMAKE_CURRENT_SOURCE_DIR}/silk>)
+
+target_link_libraries(opus PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+
+target_compile_definitions(opus PRIVATE CUSTOM_MODES OPUS_BUILD)
+
+set_property(TARGET opus PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+if(BUILD_SHARED_LIBS AND MSVC)
+  target_compile_definitions(opus PRIVATE DLL_EXPORT)
+elseif(BUILD_SHARED)
+  include(CheckCCompilerFlag)
+  check_c_compiler_flag(-fvisibility=hidden COMPILER_HAS_HIDDEN_VISIBILITY)
+  if(COMPILER_HAS_HIDDEN_VISIBILITY)
+    set_target_properties(opus PROPERTIES C_VISIBILITY_PRESET hidden)
+  endif()
+endif()
+
+add_sources_group(opus silk ${silk_sources})
+add_sources_group(opus celt ${celt_sources})
+
+if(FIXED_POINT)
+  add_sources_group(opus silk ${silk_sources_fixed})
+  target_include_directories(
+    opus
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/silk/fixed>)
+  target_compile_definitions(opus PRIVATE FIXED_POINT=1 DISABLE_FLOAT_API)
+else(FIXED_POINT)
+  add_sources_group(opus silk ${silk_sources_float})
+  target_include_directories(
+    opus
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/silk/float>)
+endif(FIXED_POINT)
+
+if(HAVE_SSE)
+
+  add_sources_group(opus celt ${celt_sources_sse})
+  add_sources_group(opus celt ${celt_sources_sse2})
+
+  if(SSE4_FLAG_SUPPORTED)
+    add_sources_group(opus celt ${celt_sources_sse4_1})
+    if(FIXED_POINT)
+      add_sources_group(opus silk ${silk_sources_fixed_sse4_1})
+    else(FIXED_POINT)
+      add_sources_group(opus silk ${silk_sources_sse4_1})
+    endif(FIXED_POINT)
+  endif(SSE4_FLAG_SUPPORTED)
+
+  if(OPUS_MAY_HAVE_SSE)
+    # Do runtime check for SIMD
+    target_compile_definitions(opus
+                               PRIVATE
+                               OPUS_X86_MAY_HAVE_SSE
+                               OPUS_X86_MAY_HAVE_SSE2
+                               OPUS_HAVE_RTCD)
+    if(SSE4_FLAG_SUPPORTED)
+      target_compile_definitions(opus PRIVATE OPUS_X86_MAY_HAVE_SSE4_1)
+    endif(SSE4_FLAG_SUPPORTED)
+
+  elseif(OPUS_PRESUME_SSE)
+    target_compile_definitions(opus
+                               PRIVATE
+                               OPUS_X86_PRESUME_HAVE_SSE
+                               OPUS_X86_PRESUME_HAVE_SSE2)
+    if(SSE4_FLAG_SUPPORTED)
+      target_compile_definitions(opus PRIVATE OPUS_X86_PRESUME_SSE4_1)
+    endif(SSE4_FLAG_SUPPORTED)
+  endif()
+endif(HAVE_SSE)
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "(armv7-a)")
+  add_sources_group(opus celt ${celt_sources_arm})
+endif()
+
+if(HAVE_NEON)
+
+  add_sources_group(opus celt ${celt_sources_arm_neon_intr})
+  add_sources_group(opus silk ${silk_sources_arm_neon_intr})
+
+  # silk arm neon depends on main_Fix.h
+  target_include_directories(
+    opus
+    PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/silk/fixed>)
```

Comment:
No need for $<BUILD_INTERFACE:> in `PRIVATE` section. You may need it in `PUBLIC` or `INTERFACE` section to distinguish between build and install stages.

Also if you don't use generator expression, you can simply add directory as `silk/fixed`:

```cmake
target_include_directories(opus PRIVATE silk/fixed)
```